json/decode.go

```diff
@@ -535,7 +535,11 @@ func (d decoder) decodeArray(b []byte, p unsafe.Pointer, n int, size uintptr, t
 		if err != nil {
 			if e, ok := err.(*UnmarshalTypeError); ok {
 				e.Struct = t.String() + e.Struct
-				e.Field = strconv.Itoa(i) + "." + e.Field
+				if e.Field != "" {
+					e.Field = strconv.Itoa(i) + "." + e.Field
+				} else {
+					e.Field = strconv.Itoa(i)
+				}
```

Comment:
What do you think about adding a method to `UnmarshalTypeError` to avoid the repeated code?
```go
e.prependField(strconv.Itoa(i))
```